react-client/src/DatasourceCredentialsForm.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+import "./App.css";
+
+interface DsCreds {
+  username: string;
+  host: string;
+  password: string;
+  dbname: string;
+}
```

Comment:
would encourage you to create an `index.d.ts` file to keep definitions like these.

Reply:
great idea